connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -296,6 +297,56 @@ public HeaderConverter newHeaderConverter(AbstractConfig config, String classPro
         return plugin;
     }
 
+
+    /**
+     * If the given configurations defines a {@link ConnectRestExtension} using the provided
+     * Class Names, return a list of new configured instances.
+     *
+     * @param klassNames         the list of Class Names of {@link ConnectRestExtension} that
+     *                           needs to instantiated and configured
+     * @param config             the configuration containing the {@link org.apache.kafka.connect.runtime.Worker}'s configuration;
+     *                           may not be null
+     * @return the instantiated and configured list of  {@link ConnectRestExtension}; EMPTY if
+     * the {@param klassNames} is NULL or EMPTY
+     * @throws ConnectException if the {@link ConnectRestExtension} implementation class could not be found
+     */
+    public List<ConnectRestExtension> newConnectRestExtensions(List<String> klassNames,
+                                                               AbstractConfig config) {
+
+        List<ConnectRestExtension> connectRestExtensions = new ArrayList<>();
+        if (klassNames == null || klassNames.isEmpty()) {
+            return connectRestExtensions;
+        }
+        for (String klassName : klassNames) {
+            ConnectRestExtension plugin;
+
+            Class<? extends ConnectRestExtension> klass;
+            try {
+                klass = pluginClass(
+                    delegatingLoader,
+                    klassName,
+                    ConnectRestExtension.class
+                );
+            } catch (ClassNotFoundException e) {
+                throw new ConnectException(
+                    "Failed to find any class that implements ConnectRestExtension and which name matches "
+                    + klassName
+                    + ", available Rest Extensions are: "
+                    + pluginNames(delegatingLoader.restExtensions())
+                );
+            }
+            plugin = newPlugin(klass);
+            if (plugin == null) {
+                throw new ConnectException(
+                    "Unable to instantiate the ConnectRestExtension '" + klassName + "'");
+            }
+            plugin.configure(config.originals());
```

Comment:
If we can make this method more generic or general purpose, then this method should probably be called within a conditional block checking whether it implements `Configurable`.

Reply:
Maybe you guys have already talked about this, but does it make sense to have a per extension prefix for the configs? similar to how it is done for Transformations?